src/cmap/auth/mongodb_oidc.ts

```diff
@@ -88,77 +88,59 @@ export class MongoDBOIDC extends AuthProvider {
       return callback(new MongoMissingCredentialsError('AuthContext must provide credentials.'));
     }
 
-    getWorkflow(credentials, (error, workflow) => {
-      if (error) {
-        return callback(error);
-      }
-      if (!workflow) {
-        return callback(
-          new MongoRuntimeError(
-            `Could not load workflow for device ${credentials.mechanismProperties.PROVIDER_NAME}`
-          )
-        );
+    let workflow;
```

Comment:
We can use maybeCallback here, or just an async iife. Then invoke the callback in a `.then`. Then we don't need to special case the synchronous throw, and when this becomes async the refactor would be just unwrapping the iife